app/helpers/fenrir_view/component_helper.rb

```diff
@@ -1,23 +1,63 @@
+# frozen_string_literal: true
+
 module FenrirView
   module ComponentHelper
+
+    # TODO: Remove?
     def render_ui(variant, slug, properties = {}, &block)
       return nil if properties == false
 
       component = FenrirView::Presenter.component_for(variant, slug, properties, validate: true)
+
+      if block_given?
+        if defined?(component.layout)
+          component.layout = self
```

Comment:
Maybe rename this method, because it's not obvious that it has side effects that are required for the line below `component.layout` beyond just passing `self` as the argument to `yield`